exporter/signalfxexporter/translation/converter.go

```diff
@@ -94,287 +83,273 @@ func NewMetricsConverter(logger *zap.Logger, t *MetricTranslator) *MetricsConver
+
+	basePoint := makeBaseDataPoint(metric)
+
+	switch metric.DataType() {
+	case pdata.MetricDataTypeNone:
+		return nil, 0
+	case pdata.MetricDataTypeIntGauge:
+		dps = convertIntDatapoints(metric.IntGauge().DataPoints(), basePoint, extraDimensions)
+	case pdata.MetricDataTypeIntSum:
+		dps = convertIntDatapoints(metric.IntSum().DataPoints(), basePoint, extraDimensions)
+	case pdata.MetricDataTypeDoubleGauge:
+		dps = convertDoubleDatapoints(metric.DoubleGauge().DataPoints(), basePoint, extraDimensions)
+	case pdata.MetricDataTypeDoubleSum:
+		dps = convertDoubleDatapoints(metric.DoubleSum().DataPoints(), basePoint, extraDimensions)
+	case pdata.MetricDataTypeIntHistogram:
+		dps = convertIntHistogram(metric.IntHistogram().DataPoints(), basePoint, extraDimensions)
+	case pdata.MetricDataTypeDoubleHistogram:
+		dps = convertDoubleHistogram(metric.DoubleHistogram().DataPoints(), basePoint, extraDimensions)
+	}
 
-		for _, series := range metric.Timeseries {
-			dimensions := make([]*sfxpb.Dimension, numLabels+len(extraDimensions))
-			copy(dimensions, extraDimensions)
-			for i := 0; i < numLabels; i++ {
-				dimension := &sfxpb.Dimension{
-					Key:   descriptor.LabelKeys[i].Key,
-					Value: series.LabelValues[i].Value,
-				}
-				dimensions[len(extraDimensions)+i] = dimension
-			}
+	if c.metricTranslator != nil {
+		dps = c.metricTranslator.TranslateDataPoints(c.logger, dps)
+	}
 
-			for _, dp := range series.Points {
-
-				var msec int64
-				if dp.Timestamp != nil {
-					msec = dp.Timestamp.Seconds*1e3 + int64(dp.Timestamp.Nanos)/1e6
-				}
-
-				sfxDataPoint := &sfxpb.DataPoint{
-					// Source field is not set by code seen at
-					// github.com/signalfx/golib/sfxclient/httpsink.go
-					Metric:     metricName,
-					MetricType: metricType,
-					Timestamp:  msec,
-					Dimensions: dimensions,
-				}
-
-				switch pv := dp.Value.(type) {
-				case *metricspb.Point_Int64Value:
-					sfxDataPoint.Value = sfxpb.Datum{IntValue: &pv.Int64Value}
-					metricDataPoints = append(metricDataPoints, sfxDataPoint)
-
-				case *metricspb.Point_DoubleValue:
-					sfxDataPoint.Value = sfxpb.Datum{DoubleValue: &pv.DoubleValue}
-					metricDataPoints = append(metricDataPoints, sfxDataPoint)
-
-				case *metricspb.Point_DistributionValue:
-					metricDataPoints = appendDistributionValues(
-						metricDataPoints,
-						sfxDataPoint,
-						pv.DistributionValue)
-				case *metricspb.Point_SummaryValue:
-					metricDataPoints = appendSummaryValues(
-						metricDataPoints,
-						sfxDataPoint,
-						pv.SummaryValue)
-				default:
-					numDroppedTimeSeries++
-					c.logger.Warn(
-						"Timeseries dropped to unexpected metric type",
-						zap.String("metric", sfxDataPoint.Metric))
-				}
+	return dps, 0
+}
 
-			}
-		}
+func labelsToDimensions(labels pdata.StringMap, extraDims []*sfxpb.Dimension) []*sfxpb.Dimension {
+	dimensions := make([]*sfxpb.Dimension, len(extraDims), labels.Len()+len(extraDims))
+	copy(dimensions, extraDims)
```

Comment:
Based on the question about the translator, we may need to do a deep copy here not just copying the pointers.

Reply:
Yeah I didn't originally write this function but I think whoever did is intentionally copying since the translator does modify structs in place via pointers instead of making copies.